lib/OpenQA/WebAPI/Controller/Main.pm

```diff
@@ -22,21 +22,24 @@ use Date::Format;
 use OpenQA::Schema::Result::Jobs;
 use OpenQA::BuildResults;
 use OpenQA::Utils;
+use Scalar::Util qw(looks_like_number);
 
 sub index {
     my ($self) = @_;
 
-    my $limit_builds    = $self->param('limit_builds')    // 3;
-    my $time_limit_days = $self->param('time_limit_days') // 14;
+    my $limit_builds = $self->param('limit_builds');
+    $limit_builds = 3 unless looks_like_number($limit_builds);
+    my $time_limit_days = $self->param('time_limit_days');
+    $time_limit_days = 14 unless looks_like_number($time_limit_days);
```

Comment:
The main problem before was that the `some value // some default` syntax also uses `some value` if it is an empty string.

Don't know whether there's a better way. Defaults for index page and overview are even different so it wouldn't make sense to outsource it to a common place.